web-ui/src/components/member-directory/MemberModal.jsx

```diff
@@ -41,6 +57,30 @@ const MemberModal = ({ member = {}, open, onSave, onClose }) => {
     });
   };
 
+  const validateRequiredInputs = () => {
+    return (editedMember.firstName?.length > 0 && editedMember.lastName?.length > 0
+       && editedMember.workEmail?.length > 0 && editedMember.title?.length > 0 && editedMember.location?.length >0 && editedMember.employeeId?.length > 0 && startDate )
+  }
+  
+  const submitMemberClick = async () => {
+    let required = validateRequiredInputs()
+    if (required) {
+      onSave(editedMember).then(() => {
+        if (isNewMember) {
+          setMember({emptyMember})
+        }
+      })
+    } else {
+      dispatch({
+        type: UPDATE_TOAST,
+        payload: {
+          severity: "error",
+          toast: "One or more required fields are empty. Check starred input fields",
+        },
+      });
+    }
+  }
+
```

Comment:
Since these are defined inside the render function, you want to create them with `useCallback`, otherwise a new method is created with every render (state changes, property changes, etc), even when it's not necessary.